Sofa/framework/Config/src/sofa/config.h.in

```diff
@@ -243,6 +243,24 @@ class DeprecatedAndRemoved {};
         "v21.06 (PR#1808)", "v21.12", \
         "replace std::cout << myaccessor by std::cout << myaccessor.ref() ")
 
+#define SOFA_ATTRIBUTE_DISABLED__NOSTYLE_TYPEMEMBER(type) \
+    SOFA_ATTRIBUTE_DISABLED( \
+        "v22.12 (PR#3357)", "v24.12", \
+        "A coding style has been introduced, to every class's member type (eg: MyObject::Real) must starts with M_ (eg: MyObject::M_Real).") \
+        DeprecatedAndRemoved type
+
+#define SOFA_ATTRIBUTE_DISABLED__TYPEMEMBER(type) \
+    SOFA_ATTRIBUTE_DISABLED( \
+        "v22.12 (PR#3357)", "v24.12", \
+        "Class's member type alias " sofa_tostring(type) " was not needed.") \
+        DeprecatedAndRemoved type
+
+#define SOFA_ATTRIBUTE_REPLACED__TYPEMEMBER(type, newtype) \
+    SOFA_ATTRIBUTE_DEPRECATED( \
+        "v22.12 (PR#3357)", "v24.12", \
```

Comment:
Well, given how invasive it will be to get rid of that as it is widely spread in plugins I would say why not. 
But If you agree on a shorter period of time I would say yes.